repos/system_upgrade/el7toel8/actors/restrictedpcisscanner/libraries/restrictedpcisscanner.py

```diff
@@ -0,0 +1,151 @@
+import json
+import os
+
+import urllib3
+
+from leapp.libraries.common.config import get_env
+from leapp.libraries.stdlib import api
+from leapp.models import RestrictedPCIDevices
+
+REQUEST_TIMEOUT = 0.5  # TODO Consider moving it to the leapp env var
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+UNSUPPORTED_DRIVER_NAMES_GLOBAL_FILE = os.path.join(
+    "/etc/leapp/files", "unsupported_driver_names.json"
+)
+UNSUPPORTED_PCI_IDS_GLOBAL_FILE = os.path.join(
+    "/etc/leapp/files", "unsupported_pci_ids.json"
+)
+UNSUPPORTED_DRIVER_NAMES_FALLBACK_FILE = os.path.join(
```

Comment:
> Can't you use relative path for that? 

I am trying to be independent from the cwd, thus, I am refering to the current directory. I tried to change it to the relative path and my tests failed. I believe because the cwd was different inside the pytest session.

> Or can this be taken from some config file?

The user shouldn't be able to affect the fallback location, because it should be100% reliable.

Reply:
For files under `/etc/leapp/files/` the absolut path should be used always. Why the CUR_DIR is used in this way? Where the "data" directory is supposed to be stored? I don't know any "data" dierctory under repository. Only thing you could have under actor are directories
- files
- libraries
- tests

There is no space for any "data" directory. And in such case, actor should not try to get there through relative paths. It should be located under "files" directory and to get the path, you should use the API functions. As well the CUR_DIR should not exist at all under the directory. You basically do not care about current directory. Fix it please. Here is the repository directory layout: https://leapp.readthedocs.io/en/latest/repository-dir-layout.html

Reply:
@ZhukovGreen btw, you are interested about the `get_actor_file_path` function if I am right, provided by the `api` library.

Reply:
Just regarding additional issues I found (the write operation in those files), if this is necessary, the file even cannot be located in the repository.

Reply:
@pirat89:
1. For files under /etc/leapp/files/ the absolute path should be used always - it is used. I think, you messed this up.
2. I renamed the actor_name/data into actro_name/files
3. I don't access the data files via a relative path. I take the absolute path of the current directory and use this as my anchor. However if you want, I can replace the CUR_DIR with something like ACTOR_ROOT = ...get_actor_file_path(...) and then do the same. However, I really don't see any difference. We can sync today about this

Reply:
Please, use get_actor_file_path https://github.com/oamg/leapp/blob/master/leapp/actors/__init__.py#L271 or any other already implemented functions from the library to do that.

Reply:
just to broadcast the decision from our cabal session, we decided do not to have a fallback mechanism, because the upgrade will be inhibited anyway if there are not needed files inside the /etc/leapp/files. This means the origin of this comment is no longer valid. Removing these vars UNSUPPORTED_DRIVER_NAMES_FALLBACK_FILE and UNSUPPORTED_PCI_IDS_FALLBACK_FILE